src/os/layer/preset/presetmenubutton.js

```diff
@@ -0,0 +1,220 @@
+goog.module('os.layer.preset.PresetMenuButton');
+goog.module.declareLegacyNamespace();
+
+const LayerPresetManager = goog.require('os.layer.preset.LayerPresetManager');
+const MenuButtonCtrl = goog.require('os.ui.menu.MenuButtonCtrl');
+const Menu = goog.require('os.ui.menu.Menu');
+const MenuItem = goog.require('os.ui.menu.MenuItem');
+const MenuItemType = goog.require('os.ui.menu.MenuItemType');
+const Module = goog.require('os.ui.Module');
+const {Presets: Metrics} = goog.require('os.metrics.keys');
+
+
+/**
+ * @const {string}
+ */
+const MENU_FLAG = 'presets';
+
+/**
+ * Preset events
+ * @enum {string}
+ */
+const EventType = {
+  REMOVE: 'remove',
+  SAVE: 'save',
+  TOGGLE_DEFAULT_TRUE: 'toggle-default-true',
+  TOGGLE_DEFAULT_FALSE: 'toggle-default-false',
+  TOGGLE_PUBLISHED_TRUE: 'toggle-published-true',
+  TOGGLE_PUBLISHED_FALSE: 'toggle-published-false'
+};
+
+/**
+ * The controller for the preset directive; make use of the MenuButtonController
+ * @unrestricted
+ */
+class Controller extends MenuButtonCtrl {
+  /**
+   * Constructor.
+   * @param {!angular.Scope} $scope
+   * @param {!angular.JQLite} $element The element
+   * @ngInject
+   */
+  constructor($scope, $element) {
+    super($scope, $element);
+
+    /**
+     * @type {boolean}
+     */
+    this['isAdmin'] = false;
+
+    this.init_();
+  }
+
+  /**
+   * @private
+   */
+  init_() {
+    this['isAdmin'] = LayerPresetManager.getInstance().isAdmin(); // requires ng-if
```

Comment:
Is this guaranteed to be resolved at the time this UI first renders?

Reply:
Yep, because it's under an ng-if, not an ng-show. The other option is to rewrite the LayerPresetManager to give access to its PresetServiceManager (which is an EventTarget) so others like this can listen to when events are thrown when Services are completed loading. This was a lot less code and eventing in the interest of finishing the other tickets; but the latter is more robust.